src/main/java/com/arpnetworking/tsdcore/model/CompositeUnit.java

```diff
@@ -0,0 +1,338 @@
+/**
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.tsdcore.model;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+import net.sf.oval.constraint.NotNull;
+
+
+/**
+ * Compound Unit used in 2g file format.
```

Comment:
If the serialized format supports all of these possible units it seems good lay some ground work for being able to use more than the current set of units.  I don't have the time right now to fully implement all of the functionality, such as transforming between units or fully replacing the existing Unit class but this seemed like a good place to start.